### DISCUSSION TITLE
Is there a better way of getting a project ID?

### DISCUSSION THREAD

**Discussion by NoobTracker** [ScratchAddons/ScratchAddons#158] [Help]
Currently, I'm searching for the last truthy element in a URL to get the project ID. The problem with that is that "editor" or "fullscreen" is truthy too. So my question is: Can you get the project ID by not using the URL?

**Comment by Explosion-Scratch** [comment]
Hmm, maybe by the API?

**Comment by NoobTracker** [comment]
@Explosion-Scratch How? I guess that API is for doing stuff like isthistaken123's stuff. Is there somewhere on the website a little thing that stores the project ID?

**Comment by apple502j** [comment]
Using fake state:
`addon.tab.traps.onceValues.fakeGUIState.projectState.projectId`

Using real state:
`addon.tab.redux.scratchGui.projectState.projectId`

Or, split pathname by `/` and find ones that are numeric.

**Comment by NoobTracker** [comment]
@apple502j Who in the world would want to use faked values?

**Comment by NoobTracker** [comment]
@apple502j Also, how does the addon API that?

**Comment by apple502j** [comment]
@NoobTracker Real redux conflicts with several extensions, so is not always reliable. For API details, see the Wiki.

**Comment by Hans5958** [comment]
Don't `document.location.pathname.split("/")[2]` is enough? 

The project ID is always in the second position after splitting the path on the slashes (example `/projects/213213223`, `/projects/213213223/`, `/projects/213213223/editor`, `/projects/213213223/fullscreen`)

**Comment by NoobTracker** [comment]
It should usually be ... but I think that there could be better ways then using the URL.

**Comment by Hans5958** [comment]
> 
> 
> It should usually be ... but I think that there could be better ways then using the URL.

Sure, sure. Just don't over-engineer things.

**Comment by NoobTracker** [comment]
Yeahyeahyeah, I just wanted to ask.

**Comment by WorldLanguages** [comment]
`document.location.pathname.split("/")[2]` is the way to go.
